src/mame/sinclair/tsconf_beta.h

```diff
@@ -0,0 +1,72 @@
+// license:BSD-3-Clause
+// copyright-holders:Andrei I. Holub
+#ifndef MAME_SINCLAIR_TSCONF_BETA_H
+#define MAME_SINCLAIR_TSCONF_BETA_H
+
+#pragma once
+
+#include "machine/wd_fdc.h"
+#include "imagedev/floppy.h"
+
+class tsconf_beta_device : public device_t
+{
+public:
+	tsconf_beta_device(const machine_config &mconfig, const char *tag, device_t *owner, u32 clock);
+
+	auto out_dos_callback() { return m_out_dos_cb.bind(); }
+	auto out_vdos_m1_callback() { return m_out_vdos_m1_cb.bind(); }
+
+	u8 status_r();
+	u8 track_r();
+	u8 sector_r();
+	u8 data_r();
+	u8 state_r();
+
+	void param_w(offs_t offset, u8 data);
+	void command_w(u8 data);
+	void track_w(u8 data);
+	void sector_w(u8 data);
+	void data_w(u8 data);
+	void turbo_w(int state);
+
+    static void beta_ioext(const device_finder<tsconf_beta_device, true> &finder, address_map &map);
+    void on_m1_w();
+    bool dos_r() { return m_dos; }
+    bool vdos_r() { return m_vdos; }
+    bool dos_io_r() { return m_dos || m_io_forced; }
+    void enable_w(bool state);
+    void fddvirt_w(u8 fddvirt);
+    void io_forced_w(bool io_forced);
```

Comment:
Mixed space/tab indentation.  Also, the way to install a device in an address map is with:

```c++
   map.m(taglike, FUNC(whatever_device::mapfunc);
```

Where `mapfunc` has a signature like `void mapfunc(address_map &)` – there are numerous examples.